### PR TITLE
Event type Localization and geofence store creation

### PR DESCRIPTION
**Pull request by mail2bishnoi**
Hi Anton,

I have implemented 'event type' localization in the events reports.

Along with this, I have created the geofence store and formatted the geofenceId to display the geofence name.

I was trying to move the 'formatGeofence' method to some centralized location i.e. in the formatter.js file, but we can not use hooks outside the react component. Please suggest how can I do this?

### CODE REVIEW

File: modern/src/MainController.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {  useDispatch, useSelector } from 'react-redux';
+import { connect } from 'react-redux';
+import { geofencesActions } from './store';
+import { useEffectAsync } from './reactHelper';
+
+
+const MainController = () => {
```
Comment (tananaev) [inline modern/src/MainController.js:7]: Where do we load devices? Maybe we can load geofences there as well.

Comment (mail2bishnoi) [reply modern/src/MainController.js:7]: Yes we can but my thought process was to keep the SocketController separate from rest of the things like Geofences, Groups, Maintenance etc. If you would like to keep all the fetches in  SocketController only then I'll move it there.

Comment (tananaev) [reply modern/src/MainController.js:7]: Maybe we need a better name then. Something like caching controller?

Comment (mail2bishnoi) [reply modern/src/MainController.js:7]: CachingController/ StoreController

**Comment by mail2bishnoi** [comment]
Renamed MainController.

File: modern/src/reports/EventReportPage.js [hunk 2]
```diff
@@ -57,8 +59,17 @@ const Filter = ({ setItems }) => {
 
 const EventReportPage = () => {
 
+  const geofences = useSelector(state => state.geofences.items);
   const [items, setItems] = useState([]);
 
+  const formatGeofence = value => {
+    if (value !== 0) {
```
Comment (tananaev) [inline modern/src/reports/EventReportPage.js:66]:
```suggestion
    if (value > 0) {
```

File: modern/src/CachingController.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import {  useDispatch, useSelector } from 'react-redux';
+import { connect } from 'react-redux';
+import { geofencesActions } from './store';
+import { useEffectAsync } from './reactHelper';
+
+
```
Comment (tananaev) [inline modern/src/CachingController.js:6]:
```suggestion

```

File: modern/src/App.js [hunk 4]
```diff
@@ -30,13 +30,16 @@ import MaintenancesPage from './settings/MaintenancesPage';
 import MaintenancePage from './settings/MaintenancePage';
 import StatisticsPage from './admin/StatisticsPage';
```
Comment (tananaev) [inline modern/src/App.js:32]: Probably remove the space here for consistency.

File: modern/src/CachingController.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  useEffectAsync(async () => {
+    if (authenticated) {
+      const response = await fetch('/api/geofences');
+      if (response.ok) {
+        dispatch(geofencesActions.update(await response.json()));
+      }
+    }
+  }, [authenticated]);
+  return null;
```
Comment (tananaev) [inline modern/src/CachingController.js:19]:
```suggestion

  return null;
```

**Comment by mail2bishnoi** [comment]
Formatting is done.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**